### PR TITLE
Compliancerule

### PR DESCRIPTION
**Pull request by wankhedeshubham**

### CODE REVIEW

File: core/src/main/java/in/aiknow/invio/constants/Rules.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class Rules {
```
Comment (sandeepmkd) [inline core/src/main/java/in/aiknow/invio/constants/Rules.java:11]: Why is this class in constants?

File: core/src/main/java/in/aiknow/invio/controller/ComplianceRuleController.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.web.bind.annotation.RestController;
+
+@Slf4j
+@Path("/compliancerule")
```
Comment (sandeepmkd) [inline core/src/main/java/in/aiknow/invio/controller/ComplianceRuleController.java:33]: compliancerules, always plural

File: core/src/main/java/in/aiknow/invio/controller/ComplianceRuleController.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+@RestController
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+@ApiModel("Api related to compliance rules")
+@ApiImplicitParams({
+    @ApiImplicitParam(name = AUTHORIZATION, required = true, paramType = "header"),
+//    @ApiImplicitParam(name = SESSION_ID_HEADER, required = true, paramType = "header")
```
Comment (sandeepmkd) [inline core/src/main/java/in/aiknow/invio/controller/ComplianceRuleController.java:40]: You can remove session id header, we don't need it anymore

File: core/src/main/java/in/aiknow/invio/entities/ComplianceRule.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Table(name = "compliance_rule")
+@Data
+@Entity
+@NoArgsConstructor
+@AllArgsConstructor
+public class ComplianceRule extends BaseObject {
```
Comment (sandeepmkd) [inline core/src/main/java/in/aiknow/invio/entities/ComplianceRule.java:20]: Generate an autogenerated compliance id

File: core/src/main/java/in/aiknow/invio/services/ComplianceRuleServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+  }
+
+  /**
+   * Get ComplianceRule by id
+   * @param id
+   * @return ComplianceRuleResponse
+   * @throws Exception
+   */
+  @Override
+  public ComplianceRuleResponse getComplianceRuleById(Long id)
```
Comment (sandeepmkd) [inline core/src/main/java/in/aiknow/invio/services/ComplianceRuleServiceImpl.java:55]: Make get and put to use the string id

File: core/src/main/java/in/aiknow/invio/services/ComplianceRuleServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+
+  /**
+   * Update ComplianceRule by id
+   * @param request
+   * @param id
+   * @return abstract response
+   * @throws Exception
+   */
+  @Override
+  @Transactional
```
Comment (sandeepmkd) [inline core/src/main/java/in/aiknow/invio/services/ComplianceRuleServiceImpl.java:79]: use @Transactional(rollbackOn=Exception.class) always

File: core/src/main/java/in/aiknow/invio/constants/Rules.java (hunk 1 above)
Comment (sandeepmkd) [reply core/src/main/java/in/aiknow/invio/constants/Rules.java:11]: Rename it to compliance rule

File: core/src/main/java/in/aiknow/invio/controller/ComplianceRuleController.java (hunk 3 above)
Comment (wankhedeshubham) [reply core/src/main/java/in/aiknow/invio/controller/ComplianceRuleController.java:40]: removed

File: core/src/main/java/in/aiknow/invio/entities/ComplianceRule.java (hunk 4 above)
Comment (wankhedeshubham) [reply core/src/main/java/in/aiknow/invio/entities/ComplianceRule.java:20]: Added

File: core/src/main/java/in/aiknow/invio/controller/ComplianceRuleController.java (hunk 2 above)
Comment (wankhedeshubham) [reply core/src/main/java/in/aiknow/invio/controller/ComplianceRuleController.java]: changed

File: core/src/main/java/in/aiknow/invio/services/ComplianceRuleServiceImpl.java (hunk 5 above)
Comment (wankhedeshubham) [reply core/src/main/java/in/aiknow/invio/services/ComplianceRuleServiceImpl.java]: changed

File: core/src/main/java/in/aiknow/invio/services/ComplianceRuleServiceImpl.java (hunk 6 above)
Comment (wankhedeshubham) [reply core/src/main/java/in/aiknow/invio/services/ComplianceRuleServiceImpl.java]: added

**Pull request merged**